assembly/src/parsers/ast/mod.rs

```diff
@@ -28,7 +28,7 @@ type LocalProcMap = BTreeMap<String, (u16, ProcedureAst)>;
 /// A program has to have a body and no exported procedures.
 #[derive(Debug, Eq, PartialEq)]
 pub struct ProgramAst {
-    pub procedures: LocalProcMap,
+    pub procedures: Vec<ProcedureAst>,
```

Comment:
minor nit: let's rename this to `local_procs` to indicate that this list contains only local procedures (i.e., procedures declared in this file).